routers/api/v1/repo/issue_comment.go

```diff
@@ -193,7 +194,7 @@ func ListIssueCommentsAndTimeline(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, &apiComments)
 }
 
-func isXRefCommentAccessible(user *user_model.User, c *models.Comment, issueRepoID int64) bool {
+func isXRefCommentAccessible(ctx stdCtx.Context, user *user_model.User, c *models.Comment, issueRepoID int64) bool {
```

Comment:
This function has repo_model.GetRepositoryByID() in it which does not use ctx as yet.

It may be helpful to add a comment to this function warning that it is not safe to be used within a session (yet)

Reply:
we have GetRepositoryByIDCtx already ... - good catch